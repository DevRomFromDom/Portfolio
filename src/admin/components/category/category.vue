<template>
    <card slim>
        <edit-line slot="title" v-model="categoryTitle" :editModeByDefault="empty"  @remove="$emit('remove', $event)"/>
        <template slot="content">
            <ul class="skills" v-if="empty === false">
                <li class="item" v-for="item in skills" :key="item.id">
                    <skill :skill="item" @remove="$emit('remove-skill', $event)" @approve="$emit('edit-skill',$event)"/>
                </li>
            </ul>
            <div class="buttom-line">
                <skill-add-line :blocked="empty" />
            </div>
        </template>
    </card>
</template>

<script>
import card from "../card/card.vue";
import EditLine from "../editLine/editLine.vue";
import skill from "../skill/skill.vue";
import skillAddLine from "../skillAddLine/skillAddLine.vue";

export default {
    props: {
        empty: Boolean,
        title: { type: String, default:"" },
        skills: { type: Array, default: () => [] },
    },
    components: { card, EditLine, skill, skillAddLine },
    data() {
        return {
            categoryTitle: this.title,
        };
    }
}
</script>

<style lang="postcss" scoped>
.skills {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
}
.item {
    margin-bottom: 30px;
    &:last-child {
        margin-bottom: 0;
    }
}
.buttom-line {
    padding-top: 70px;
    margin-top: auto;
    padding-left: 25%;
}
</style>
