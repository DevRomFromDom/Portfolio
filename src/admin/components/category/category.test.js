import category from "./category.vue"
import {shallowMount} from "@vue/test-utils"





it("Cayegory has been mounted", async ()=>{
    const wrapper = shallowMount(category, {propsData:{
        id: 1,
        title: "title",
        skills:[{id:1, title: "skill-Title", percent: 25},{id:2, title: "skill-Title", percent: 65},{id:3, title: "skill-Title", percent: 35},]
    }})
    expect(wrapper.element).toMatchSnapshot();
})