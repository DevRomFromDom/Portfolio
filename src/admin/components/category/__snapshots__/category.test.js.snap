// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cayegory has been mounted 1`] = `
<card-stub
  slim=""
  title=""
>
  <edit-line-stub
    errortext=""
    value="title"
  />
   
  <template>
    <ul
      class="skills"
    >
      <li
        class="item"
      >
        <skill-stub
          skill="[object Object]"
        />
      </li>
      <li
        class="item"
      >
        <skill-stub
          skill="[object Object]"
        />
      </li>
      <li
        class="item"
      >
        <skill-stub
          skill="[object Object]"
        />
      </li>
    </ul>
     
    <div
      class="buttom-line"
    >
      <skill-add-line-stub />
    </div>
  </template>
</card-stub>
`;
